pkg/controller/clusterdeployment/clusterdeployment_controller_test.go

```diff
@@ -1482,6 +1482,26 @@ func TestClusterDeploymentReconcile(t *testing.T) {
 				assert.Equal(t, "manual-resourcegroup", deprovision.Spec.Platform.Azure.ResourceGroupName)
 			},
 		},
+		{
+			name: "set ProvisionStoppedCondition when install restarts limit has reached",
+			existing: []runtime.Object{
+				func() runtime.Object {
+					cd := testClusterDeployment()
+					cd.Status.InstallRestarts = 3
+					cd.Spec.LimitInstallRestarts = new(int)
+					*cd.Spec.LimitInstallRestarts = 2
```

Comment:
Assuming that `LimitInstallRestarts` is changed to be an `*int32`,
```suggestion
					cd.Spec.LimitInstallRestarts = pointer.Int32Ptr(2)
```